packages/async-flow/test/replay-membrane-eventual.test.js

```diff
@@ -0,0 +1,84 @@
+// eslint-disable-next-line import/order
+import {
+  test,
+  getBaggage,
+  annihilate,
+  nextLife,
+} from './prepare-test-env-ava.js';
+
+import { Fail } from '@endo/errors';
+import { prepareVowTools } from '@agoric/vow';
+import { E } from '@endo/eventual-send';
+// import E from '@agoric/vow/src/E.js';
+import { makeHeapZone } from '@agoric/zone/heap.js';
+import { makeVirtualZone } from '@agoric/zone/virtual.js';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { prepareLogStore } from '../src/log-store.js';
+import { prepareBijection } from '../src/bijection.js';
+import { makeReplayMembrane } from '../src/replay-membrane.js';
+
+const watchWake = _vowish => {};
+const panic = problem => Fail`panic over ${problem}`;
+
+/**
+ * @param {Zone} zone
+ */
+const preparePingee = zone =>
+  zone.exoClass('Pingee', undefined, () => ({}), {
+    ping() {},
+  });
+
+/**
+ * @typedef {ReturnType<ReturnType<preparePingee>>} Pingee
+ */
+
+/**
+ * @param {any} t
+ * @param {Zone} zone
+ */
+const testFirstPlay = async (t, zone) => {
+  const vowTools = prepareVowTools(zone);
+  const makeLogStore = prepareLogStore(zone);
+  const makeBijection = prepareBijection(zone);
+  const makePingee = preparePingee(zone);
+
+  const log = zone.makeOnce('log', () => makeLogStore());
+  const bij = zone.makeOnce('bij', makeBijection);
+
+  const mem = makeReplayMembrane(log, bij, vowTools, watchWake, panic);
+
+  t.deepEqual(log.dump(), []);
+
+  /** @type {Pingee} */
+  const pingee = zone.makeOnce('pingee', () => makePingee());
+  /** @type {Pingee} */
+  const guestPingee = mem.hostToGuest(pingee);
+  t.deepEqual(log.dump(), []);
+
+  await t.throwsAsync(() => E(guestPingee).ping(), {
+    message:
+      'panic over "[Error: guest eventual send not yet supported: \\"[Alleged: Pingee guest wrapper]\\".ping([]) -> \\"[Promise]\\"]"',
+  });
+
+  t.deepEqual(log.dump(), []);
```

Comment:
I'd like to capture in a test that the handler is not synchronously invoked, and this can have effects on the persisted log before the panic occurs.

```suggestion
  const pingTestSendResult = t.throwsAsync(() => E(guestPingee).ping("send"), {
    message:
      'panic over "[Error: guest eventual send not yet supported: \\"[Alleged: Pingee guest wrapper]\\".ping(["send"]) -> \\"[Promise]\\"]"',
  });
  
  guestPingee.ping("call");
  
  await pingTestSendResult;

  t.deepEqual(log.dump(), [
    ['checkCall', pingee, 'ping', ["call"], 1],
    ['doReturn', 1, undefined],
  ]);
```

Reply:
Used your code -- thanks!

Done.